refactor: remove stacks ui part 3

fbwoolf opened the pull request:
> Try out this version of Leather — [download extension builds](https://github.com/leather-wallet/extension/actions/runs/6898555777).<!-- Sticky Header Marker -->

PR to remove @stacks/ui (issue Part 3) ...lots of files changed here so need to test and open to feedback. I setup the `ui` folder and started migrating new brand icons and some other things there to separate out what might be extracted out to share later. Most of the complexity here was having to address all the asset and tx list icons and needing to remove the `typography.tsx` file. I ended up only keeping/relocating the `Caption` component here. I was able to remove `Body`, `Title`, and `Text` ...but there might be some minor UI differences.

kyranjamie commented on public/assets/base.css:
```diff
@@ -42,3 +42,13 @@
   font-style: normal;
 }
+
+@keyframes spin {
+  0% {
+    transform: rotate(0deg);
+  }
+
+  100% {
+    transform: rotate(360deg);
+  }
+}
```
Keyframes should be described in Panda. Why add it straight to css here? Outside of font definitions, there should be no css here.

pete-watters replied:
On yesterdays call I said I didn't understand how to do some things with panda yet. Once of them is properly configuring it to provide the `base.css`. Do you have any ideas about how to do that? 

Asking as I was working on this [task](https://github.com/leather-wallet/extension/issues/4398) and came up with this [WIP PR](https://github.com/leather-wallet/extension/pull/4494). 

For that bug:
- when the app is loaded the BG colour flashes white before the them BG colour gets set properly
- we can try set BG to the theme BG but we don't know what it is until Panda provides it
- I was trying to set the colour in the `base.css` and it works but not in all cases

Panda has different [layers](https://panda-css.com/docs/concepts/cascade-layers) but I couldn't understand how to extend and configure them properly. For example, ideally we could have our fonts loaded in the `@layer base` 

Maybe you understand better how that works @kyranjamie ?

fbwoolf replied:
I did try to import `keyframes` from leather-styles here, but I don't think it worked. I can try it again thou.

fbwoolf replied:
I see now `spin` is built into panda. 👍

kyranjamie replied:
Afaik we basically shouldn't ever need to touch the base.css. If we need global styles, panda has it's method for adding these [imagine that's where the dark mode stuff would go](https://panda-css.com/docs/migration/styled-components#global-styles).

kyranjamie replied:
The layers thing is the only direct css file we need to add re: panda, and it is just to pick and choose which panda-generated styles we wanna add.

kyranjamie commented on src/app/common/transactions/stacks/transaction.utils.ts:
```diff
@@ -140,3 +141,7 @@ export function getEstimatedConfirmationTime(
   return `~${arrivesIn / 60} min`;
 }
+
+export function isPendingTx(tx: MempoolTransaction): boolean {
+  return tx.tx_status === 'pending';
+}
```
A `MempoolTransaction` is always pending, no? 

Doesn't this fn want to allow a wider type (all txs), and infer `MempoolTransaction` from it? 

```suggestion
export function isPendingTx(tx: AllTxTypes): tx is MempoolTransaction {
  return tx.tx_status === 'pending';
}
```

kyranjamie commented on src/app/common/transition.ts:
```diff
@@ -0,0 +1 @@
+export const transition = 'all 0.2s cubic-bezier(0.23, 1, 0.32, 1)';
```
Use panda for this https://panda-css.com/docs/theming/tokens#easings

fbwoolf replied:
Ok, thx, will relocate.

kyranjamie commented on src/app/components/address-displayer/form-address-displayer.tsx:
```diff
@@ -15,7 +14,7 @@ export function FormAddressDisplayer({ address, ...rest }: FormAddressDisplayerP
       justifyContent="end"
       mr="-8px"
-      fontSize="16px"
+      fontSize="lg"
```
Here we should probably use a `textStyle` instead?

pete-watters replied:
We should set `textStyle` on the `AddressDisplayerLayout` component a few levels below this

fbwoolf replied:
Ok, will check this.

kyranjamie commented on src/app/components/drawer/base-drawer.tsx:
```diff
@@ -56,20 +55,19 @@ export const BaseDrawer = memo((props: BaseDrawerProps) => {
     <Flex
       display={isShowing ? 'flex' : 'none'}
-      bg={`rgba(0,0,0,0.${isShowing ? 4 : 0})`}
+      bg="overlay"
```
Removed conditionality here?

fbwoolf replied:
I don't think this was ever nec bc of the display conditional above it. It seemed fine to remove it.

pete-watters replied:
For stuff like this I think we should remove it as you did, then re-add it if it's noticed as a bug. That will help us simplify things as we go too

kyranjamie commented on src/app/components/drawer/base-drawer.tsx:
```diff
@@ -93,11 +91,11 @@ export const BaseDrawer = memo((props: BaseDrawerProps) => {
+        bg="accent.background-primary"
+        borderTopLeftRadius="lg"
+        borderTopRightRadius="lg"
+        borderBottomLeftRadius={[0, 'lg', 'lg', 'lg']}
+        borderBottomRightRadius={[0, 'lg', 'lg', 'lg']}
         position="relative"
```
```suggestion
        borderBottomLeftRadius={[0, 'lg']}
        borderBottomRightRadius={[0, 'lg']}
        position="relative"
```

kyranjamie commented on src/app/components/item-hover.tsx:
```diff
@@ -30,9 +31,10 @@ type HoverBind = ReturnType<typeof useHover>[1];
 type FocusBind = ReturnType<typeof useFocus>[1];
 
 interface DefaultSpreadProps extends HoverBind, FocusBind {
+  bg: 'transparent';
```
What's the point of a string literal here? If it can only be one value, it doesn't need to be a prop

fbwoolf replied:
I had to set this now or it was white and hid the content (I think that was it). I can check it again, but removing @stacks/ui here did break it.

pete-watters replied:
Lets collect stuff like this in a list and we can tackle that in future after we've done the big job of removing the library

kyranjamie commented on src/app/features/stacks-transaction-request/contract-deploy-details/contract-deploy-details.tsx:
```diff
@@ -43,22 +44,21 @@ function ContractCodeSection() {
-    <Caption
-      as="button"
+    <styled.button
+      bg={isActive ? 'accent.component-background-hover' : 'transparent'}
```
probably doesn't work? https://panda-css.com/docs/guides/dynamic-styling#runtime-values

fbwoolf replied:
I thought I did check it, but will double check bc doesn't seem like it should.

fbwoolf replied:
This does toggle correctly... 🤔 

<img width="220" alt="Screen Shot 2023-11-09 at 9 26 31 AM" src="https://github.com/leather-wallet/extension/assets/6493321/1bd57d39-c723-4b65-8e87-59e481f4cca4">

fbwoolf replied:
Dynamic styling does work when using a token.

fbwoolf replied:
https://panda-css.com/docs/guides/dynamic-styling#using-token

kyranjamie reviewed (commented):
Let's also get @mica000 or @fabric-8's eyes on this as there may be obvious design discrepancies introduced

pete-watters commented:
> Let's also get @mica000 or @fabric-8's eyes on this as there may be obvious design discrepancies introduced

I don't think we should ask for a design review yet as there's plenty more work to do. 

- I think we should have a single base branch - this one. 
- Then we can review and merge the other tasks into here
- Once we are happy and think everything looks OK then we can get a design review

Other tasks for reference:
- https://github.com/leather-wallet/extension/issues/4380
- https://github.com/leather-wallet/extension/issues/4381
- https://github.com/leather-wallet/extension/issues/4382
- https://github.com/leather-wallet/extension/issues/4383
- https://github.com/leather-wallet/extension/issues/4384
-

pete-watters commented on src/app/components/available-balance.tsx:
```diff
@@ -16,21 +13,20 @@ export function AvailableBalance(props: { balance: Money; balanceTooltipLabel?:
+          Available balance
+        </styled.span>
+        <Tooltip label={balanceTooltipLabel} placement="top">
+          <Box>
```
I don't think we need `Box`'s to wrap `<Icon`s

fbwoolf replied:
This is always nec for our tooltip to work. I've never totally understood why, but can look into it. If I don't have a `Box` wrapping it, then you don't see the tooltip.

pete-watters replied:
OK no problem, lets leave it so, thanks!

pete-watters commented on src/app/components/bitcoin-contract-entry-point/bitcoin-contract-entry-point-layout.tsx:
```diff
@@ -30,21 +28,23 @@ export function BitcoinContractEntryPointLayout(props: BitcoinContractEntryPoint
+        <HStack alignItems="center" justifyContent="space-between" width="100%">
+          <styled.span textStyle="caption.02">{caption}</styled.span>
+          <Flex>
+            {isLoading ? '' : <styled.span textStyle="caption.02">{usdBalance}</styled.span>}
```
Maybe we should have a `LoadingSpinner` here too when `isLoading` ?

fbwoolf replied:
Yeah, possibly, this is code implemented by the DLC.Link team, but I can try to check it.

pete-watters commented on src/app/common/transactions/bitcoin/utils.ts:
```diff
@@ -9,6 +9,27 @@ import { satToBtc } from '@app/common/money/unit-conversion';
+  if (tx.status.confirmed) return 'success';
+  return 'pending';
+};
+
```
Nice work on this 👍 I was confused by the legacy `ColorsStringLiteral`. 

Maybe we can work with design on these colours to update them and have new semantic tokens for them? (We can make a task for it). 

- `pending` 
- `feedback-error` etc. 

I found some other hardcoded colours in `password-field.utils.ts` also

pete-watters commented on src/app/common/utils.ts:
```diff
@@ -307,3 +307,27 @@ export function removeTrailingNullCharacters(s: string) {
+function truncateHex(hex: string, offset = 5): string {
+  return `${hex.substring(0, offset + 2)}…${hex.substring(hex.length - offset)}`;
+}
+
```
Lets do a find/replace of all imports of this

pete-watters commented on src/app/components/bitcoin-transaction-item/bitcoin-transaction-icon.tsx:
```diff
@@ -1,32 +1,42 @@
+  return <ArrowUpIcon size="13px" />;
+}
+
+interface TransactionIconProps extends CircleProps {
```
Our code confuses me sometimes as here we are saying that:
- `interface TransactionIconProps extends CircleProps` 
- Then we give those props to `BitcoinTransactionIcon`
- But when we call the component
    - we only do it once
    - we only pass `transaction` and `address` - `<BitcoinTransactionIcon transaction={transaction} btcAddress={bitcoinAddress} />` 

So it seems there is no need to extend `CircleProps` or to spread any extra props here

pete-watters replied:
Yeah it makes sense not to change too much. I just thought this was weird when refactoring. Often I was making changes to appease `extends Props` that weren't really used then

pete-watters commented on src/app/components/bitcoin-transaction-item/bitcoin-transaction-inscription-icon.tsx:
```diff
@@ -17,8 +19,8 @@ function InscriptionIcon({ inscription, ...rest }: { inscription: SupportedInscr
         <Circle
-          bg={color('accent')}
-          color={color('bg')}
+          bg="stacks"
```
Maybe this can be a semantic color? or is `stacks` a specific color?

fbwoolf replied:
I did create `stacks` as a token to use to replace the old @stacks/ui Stacks `brand` and `accent`, which were the same purple.

pete-watters commented on src/app/components/crypto-assets/stacks/components/stacks-asset-avatar.tsx:
```diff
@@ -9,29 +10,29 @@ interface StacksAssetAvatarProps extends BoxProps {
-    return (
-      <img height={imageDimension} width={imageDimension} src={encodeURI(imageCanonicalUri)} />
-    );
+    return <img height={`${size}px`} width={`${size}px`} src={encodeURI(imageCanonicalUri)} />;
```
👏

pete-watters commented on src/app/components/drawer/components/header-action-button.tsx:
```diff
@@ -13,22 +14,23 @@ export function HeaderActionButton(props: HeaderActionButtonProps) {
       borderRadius="8px"
-      color={token('colors.brown.12')}
+      color="accent.action-primary-default"
+      height="36px"
```
We have `36px` in the code a lot, I wonder if we need to add some tokens for size?

pete-watters replied:
Yes I think we should, we can make a new task to do it

pete-watters commented on src/app/components/header.tsx:
```diff
@@ -1,21 +1,20 @@
 import { useMemo } from 'react';
-import { FiArrowLeft } from 'react-icons/fi';
 import { useLocation, useNavigate } from 'react-router-dom';
 
```
Great work here 👍

pete-watters commented on src/app/components/loading-rectangle.tsx:
```diff
@@ -1,11 +1,11 @@
-      backgroundImage="linear-gradient(90deg, rgba(219,219,219,1) 0%, rgba(192,192,247,0.5) 35%, rgba(219,219,219,1) 100%)"
-      backgroundColor="rgb(219,219,219)"
       animation="shine 5s infinite linear"
+      backgroundColor="rgb(219,219,219)"
```
Maybe we can ask the designers to give us color codes for this?

fbwoolf replied:
There might be one in Figma already, will check.

pete-watters commented on src/app/components/transaction/token-transfer-icon.tsx:
```diff
@@ -0,0 +1,28 @@
+  if ('is_unanchored' in tx && tx.is_unanchored)
+    return (
+      <MicroblockIcon
+        bg={getColorFromTx(tx)}
```
I think we should migrate the colors from `getColorFromTx` to our tokens

fbwoolf replied:
Not sure what you mean here, they are tokens? 🤔 Look in `semantic-tokens.ts` file here.

pete-watters replied:
Sorry thats my mistake ✋  I didn't see them there

fbwoolf commented on src/app/components/transaction-item/transaction-item.layout.tsx:
```diff
@@ -1,15 +1,16 @@
-  txStatus?: JSX.Element;
-  belowCaptionEl?: JSX.Element;
-  children?: JSX.Element;
+  txCaption: React.JSX.Element;
+  txTitle: React.JSX.Element;
+  txValue: React.JSX.Element;
+  txIcon?: React.JSX.Element;
+  txStatus?: React.JSX.Element;
+  belowCaptionEl?: React.JSX.Element;
+  children?: React.JSX.Element;
```
I was actually debating about that, thx.

pete-watters commented:
> > Let's also get @mica000 or @fabric-8's eyes on this as there may be obvious design discrepancies introduced
> 
> I don't think we should ask for a design review yet as there's plenty more work to do.
> 
> * I think we should have a single base branch - this one.
> * Then we can review and merge the other tasks into here
> * Once we are happy and think everything looks OK then we can get a design review
> 
> Other tasks for reference:
> 
> * [Remove `@stacks/ui` part 1 #4380](https://github.com/leather-wallet/extension/issues/4380)
> * [Remove `@stacks/ui` part 2 #4381](https://github.com/leather-wallet/extension/issues/4381)
> * [Remove `@stacks/ui` part 3 #4382](https://github.com/leather-wallet/extension/issues/4382)
> * [Remove `@stacks/ui` part 4 #4383](https://github.com/leather-wallet/extension/issues/4383)
> * [Remove `@stacks/ui` part 5 #4384](https://github.com/leather-wallet/extension/issues/4384)

We discussed this yesterday and agreed that Fara and I will wrap up our work and get that tested and merged:
-  https://github.com/leather-wallet/extension/issues/4382
- https://github.com/leather-wallet/extension/issues/4383

pete-watters commented on src/app/components/address-displayer/address-displayer.layout.tsx:
```diff
@@ -8,9 +8,10 @@ export function AddressDisplayerLayout({ isEven, ...props }: AddressDisplayerLay
       color={isEven ? 'accent.text-primary' : 'accent.text-subdued'}
-      fontFamily="Fira Code"
-      mr="tight"
+      fontSize="16px"
```
Maybe we can add the `fontSize` +  `lineHeight` into the `typography.ts` file? 

`"mono"` is only used here and in the `AppVersionLabel`

fbwoolf replied:
Yeah, but in the `AppVersionLabel` the font size is different, so I am thinking that is why it was left out of `mono` originally?

fbwoolf replied:
Going to add more info to the `mono` token and break up into multiple to use with two different sizes.

pete-watters replied:
Great stuff 👍

pete-watters commented on src/app/components/bitcoin-fees-list/components/fees-list-item.tsx:
```diff
@@ -1,4 +1,4 @@
-import { Box, transition } from '@stacks/ui';
+import { Box } from '@stacks/ui';
```
If we update this `Box` to be from `leather-styles` maybe we don't need to use `token` anymore?

fbwoolf replied:
Yep, true, I can go back and do. Was just doing some changes quickly when not on my list of files in the issue. I felt like the PR was getting really big.

pete-watters commented on src/app/pages/fund/components/fund-account-tile.tsx:
```diff
@@ -1,4 +1,4 @@
-import { Box, Stack, color, transition } from '@stacks/ui';
+import { Box, Stack, color } from '@stacks/ui';
```
Maybe we can swap these components out too? I don't want it to be a never-ending task either though

pete-watters commented on src/app/pages/fund/components/fast-checkout-badge.tsx:
```diff
@@ -20,7 +20,7 @@ export function FastCheckoutBadge() {
     >
       <FiZap color="#008051" size="12px" strokeWidth="2.5px" />
-      <Caption color="#008051" fontWeight={500} variant="c2">
+      <Caption color="#008051" fontWeight={500}>
```
@mica000 @fabric-8 : Can you please help us update this to our new colours? For the `Fast Checkout`, what colour should we use for `#008051`

![Screenshot 2023-11-09 at 09 52 20](https://github.com/leather-wallet/extension/assets/2938440/5412ace1-e631-4345-9c38-59ca013a7514)

pete-watters commented on src/app/pages/fund/components/zero-percent-fees-badge.tsx:
```diff
@@ -20,7 +20,7 @@ export function ZeroPercentFeesBadge() {
     >
       <FiStar color="#F59300" size="12px" strokeWidth="2.5px" />
-      <Caption color="#FFA953" fontWeight={500} variant="c2">
+      <Caption color="#FFA953" fontWeight={500}>
```
@fabric-8 @mica000 : Can you please help with this too? 

What new colour should we use for `#F59300` and `#FFA953` 

It's shown in this `0% fees` here:
![Screenshot 2023-11-09 at 09 54 26](https://github.com/leather-wallet/extension/assets/2938440/5451d3f1-eded-4982-93f7-2d9ef9ac2624)

pete-watters replied:
Michelly gave the colours for this here: https://www.figma.com/file/2MLHeIeL6XPVi3Tc2DfFCr/%E2%9D%96-Leather-%E2%80%93-Design-System?type=design&node-id=1608-66&mode=design&t=2wIcBIcHNQLISVon-0 

I'll update soon

fbwoolf replied:
These colors are already available as `warning.label` and `success.label`. I will address this since was in my original PR.

pete-watters replied:
OK great, thank you!

pete-watters reviewed (commented):
Great work here @fbwoolf 👏 The code LGTM but I didn't approve yet as I haven't tested the UI. I'll prep my branch based on this and then test it

https://protectourcoast.co.uk/wp-content/uploads/2023/02/Screenshot-2023-02-07-at-18.06.37.png

pete-watters commented on src/app/components/crypto-assets/stacks/fungible-token-asset/stacks-fungible-token-asset-item.tsx:
```diff
@@ -1,7 +1,7 @@
-import { BoxProps } from '@stacks/ui';
 import { forwardRefWithAs } from '@stacks/ui-core';
```
I don't know / understand what this `forwardRefWithAs` does. Maybe we can remove/ migrate it in a separate task?

fbwoolf replied:
> I don't know / understand what this `forwardRefWithAs` does. Maybe we can remove/ migrate it in a separate task? 

It allows the 'as' prop. I have removed it in all the files listed for part 3. We should remove as we go.

fbwoolf commented:
> Great work here @fbwoolf 👏 The code LGTM but I didn't approve yet as I haven't tested the UI. I'll prep my branch based on this and then test it
> 
> 
> 
> <img width="435" alt="Screenshot-2023-02-07-at-18 06 37" src="https://github.com/leather-wallet/extension/assets/2938440/10d21cc8-0804-4ad1-a211-97e78595b420">
> 
> 

Great, going to finish up addressing code comments today so will push a few more changes here.

markmhendrickson commented:
@fbwoolf and @pete-watters to make some final changes then assign to @fabric-8 and @mica000 for thorough design QA

fbwoolf commented:
@pete-watters I've been working today to fix things/tests. I've just made commits to the branch here trying to get all the tests to pass. I did port over the `CodeBlock` from the old library temporarily but if you have a better idea feel free to remove. I'm not sure what is causing the unit tests to fail, but there might be a bad import somewhere to track down? There appears to be a final integration test failing with the bitcoin contracts code, but I didn't get a chance to tackle it. The only thing I found in the UI still broken is the password strength indicator. If you don't get to this stuff, I'll pick it back up tomorrow morning!

fbwoolf commented:
@mica000 and @fabric-8, you might dive in and start to test the build here?

pete-watters commented:
> @pete-watters I've been working today to fix things/tests. I've just made commits to the branch here trying to get all the tests to pass. I did port over the `CodeBlock` from the old library temporarily but if you have a better idea feel free to remove. I'm not sure what is causing the unit tests to fail, but there might be a bad import somewhere to track down? There appears to be a final integration test failing with the bitcoin contracts code, but I didn't get a chance to tackle it. The only thing I found in the UI still broken is the password strength indicator. If you don't get to this stuff, I'll pick it back up tomorrow morning!

That's great, thanks @fbwoolf 👍  I'll pick up here and try fix the password indicator. 

I didn't move `CodeBlock` as we only use it twice and I think we could just use a `<pre` tag for the same effect so I used a `<styled.pre`. We only really use it in `src/app/features/stacks-transaction-request/contract-deploy-details/contract-deploy-details.tsx` as the other place is an error boundary. 

Looking at the code you added and I see it does give us some extras like line-number etc. so good to keep it if we want that.

pete-watters commented on src/app/components/event-card.tsx:
```diff
@@ -26,7 +26,7 @@ export function EventCard(props: EventCardProps): React.JSX.Element {
           {/* TODO: What does this do? */}
           {actions && (
-            <styled.button width="24px" position="absolute" right={0}>
+            <styled.button width="24px" position="absolute" right={0} type="button">
```
Did we ever find out what this does? Maybe we can remove the `TODO: What does this do?` above?

fbwoolf replied:
I'll double check this today.

pete-watters commented on src/app/components/bitcoin-fees-list/components/fees-list-item.tsx:
```diff
@@ -32,6 +31,7 @@ export function FeesListItem({
       px="space.04"
       py="space.06"
       transition="transition"
+      type="button"
```
It's a bit late to consider this now, but when we start on the UI library lets make a generic `Button` / update `LeatherButton` so we can replace all these `styled.button` and enforce `type=button` as a standard

fbwoolf replied:
Yep, agree!

pete-watters replied:
https://github.com/leather-wallet/extension/issues/4523

pete-watters commented on src/app/pages/bitcoin-contract-request/components/bitcoin-contract-request-actions.tsx:
```diff
@@ -24,10 +24,7 @@ export function BitcoinContractRequestActions({
-      // - can we pass it with shorthand?
-      borderTop="1px solid"
-      borderColor="border-default"
+      borderTop="default"
```
Thanks, you're the best! 👏

pete-watters commented:
Looking at the failing unit tests and they fail complaining about `nanoid` in `redux-devtools`:
```
 FAIL  src/app/query/bitcoin/address/utxos-by-address.spec.tsx [ src/app/query/bitcoin/address/utxos-by-address.spec.tsx ]
Error: require() of ES Module /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/node_modules/nanoid/non-secure/index.js from /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/lib/cjs/index.js not supported.
Instead change the require of /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/node_modules/nanoid/non-secure/index.js in /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/lib/cjs/index.js to a dynamic import() which is available in all CommonJS modules.
```

I can see that got updated when re-adding `prism` [here](https://github.com/leather-wallet/extension/pull/4490/commits/c865dd40f5ebb37b30079c7dfd16b8da89d24b6c). I'm trying to think how we can solve it. We don't use `nanoid` 

Here's a [SO](https://stackoverflow.com/questions/72597602/nanoid4-in-codecept-error-err-require-esm-require-of-es-module)

pete-watters commented:
> Looking at the failing unit tests and they fail complaining about `nanoid` in `redux-devtools`:
> 
> ```
> ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Suites 4 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
> 
>  FAIL  src/app/store/transactions/post-conditions.spec.ts [ src/app/store/transactions/post-conditions.spec.ts ]
> Error: require() of ES Module /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/node_modules/nanoid/non-secure/index.js from /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/lib/cjs/index.js not supported.
> Instead change the require of /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/node_modules/nanoid/non-secure/index.js in /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/lib/cjs/index.js to a dynamic import() which is available in all CommonJS modules.
>  ❯ Object.<anonymous> node_modules/@redux-devtools/utils/lib/cjs/index.js:27:18
>  ❯ Object.<anonymous> node_modules/@redux-devtools/remote/lib/cjs/devTools.js:14:14
>  ❯ Object.<anonymous> node_modules/@redux-devtools/remote/lib/cjs/index.js:18:41
>  ❯ async /Users/petew/src/leather/extension/src/app/store/index.ts:2:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/networks/networks.hooks.ts:6:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/common/api-clients.hooks.ts:9:31
>  ❯ async /Users/petew/src/leather/extension/src/app/query/stacks/tokens/fungible-tokens/fungible-token-metadata.query.ts:2:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/transactions/post-conditions.hooks.ts:4:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/transactions/post-conditions.spec.ts:4:31
> 
> ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
> Serialized Error: { code: 'ERR_REQUIRE_ESM' }
> ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/4]⎯
> 
>  FAIL  src/app/query/bitcoin/address/transactions-by-address.spec.ts [ src/app/query/bitcoin/address/transactions-by-address.spec.ts ]
> Error: require() of ES Module /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/node_modules/nanoid/non-secure/index.js from /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/lib/cjs/index.js not supported.
> Instead change the require of /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/node_modules/nanoid/non-secure/index.js in /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/lib/cjs/index.js to a dynamic import() which is available in all CommonJS modules.
>  ❯ Object.<anonymous> node_modules/@redux-devtools/utils/lib/cjs/index.js:27:18
>  ❯ Object.<anonymous> node_modules/@redux-devtools/remote/lib/cjs/devTools.js:14:14
>  ❯ Object.<anonymous> node_modules/@redux-devtools/remote/lib/cjs/index.js:18:41
>  ❯ async /Users/petew/src/leather/extension/src/app/store/index.ts:2:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/networks/networks.hooks.ts:6:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/common/api-clients.hooks.ts:9:31
>  ❯ async /Users/petew/src/leather/extension/src/app/query/bitcoin/address/transactions-by-address.query.ts:2:31
>  ❯ async /Users/petew/src/leather/extension/src/app/query/bitcoin/address/transactions-by-address.hooks.ts:4:31
>  ❯ async /Users/petew/src/leather/extension/src/app/query/bitcoin/address/transactions-by-address.spec.ts:2:31
> 
> ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
> Serialized Error: { code: 'ERR_REQUIRE_ESM' }
> ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/4]⎯
> 
>  FAIL  src/app/query/bitcoin/address/utxos-by-address.spec.tsx [ src/app/query/bitcoin/address/utxos-by-address.spec.tsx ]
> Error: require() of ES Module /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/node_modules/nanoid/non-secure/index.js from /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/lib/cjs/index.js not supported.
> Instead change the require of /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/node_modules/nanoid/non-secure/index.js in /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/lib/cjs/index.js to a dynamic import() which is available in all CommonJS modules.
>  ❯ Object.<anonymous> node_modules/@redux-devtools/utils/lib/cjs/index.js:27:18
>  ❯ Object.<anonymous> node_modules/@redux-devtools/remote/lib/cjs/devTools.js:14:14
>  ❯ Object.<anonymous> node_modules/@redux-devtools/remote/lib/cjs/index.js:18:41
>  ❯ async /Users/petew/src/leather/extension/src/app/store/index.ts:2:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/networks/networks.hooks.ts:6:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/networks/networks.selectors.ts:5:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks.ts:7:31
>  ❯ async /Users/petew/src/leather/extension/src/app/query/bitcoin/address/utxos-by-address.hooks.ts:2:31
>  ❯ async /Users/petew/src/leather/extension/src/app/query/bitcoin/address/utxos-by-address.spec.tsx:3:31
> 
> ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
> Serialized Error: { code: 'ERR_REQUIRE_ESM' }
> ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/4]⎯
> 
>  FAIL  src/app/query/bitcoin/ordinals/inscriptions.spec.ts [ src/app/query/bitcoin/ordinals/inscriptions.spec.ts ]
> Error: require() of ES Module /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/node_modules/nanoid/non-secure/index.js from /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/lib/cjs/index.js not supported.
> Instead change the require of /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/node_modules/nanoid/non-secure/index.js in /Users/petew/src/leather/extension/node_modules/@redux-devtools/utils/lib/cjs/index.js to a dynamic import() which is available in all CommonJS modules.
>  ❯ Object.<anonymous> node_modules/@redux-devtools/utils/lib/cjs/index.js:27:18
>  ❯ Object.<anonymous> node_modules/@redux-devtools/remote/lib/cjs/devTools.js:14:14
>  ❯ Object.<anonymous> node_modules/@redux-devtools/remote/lib/cjs/index.js:18:41
>  ❯ async /Users/petew/src/leather/extension/src/app/store/index.ts:2:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/networks/networks.hooks.ts:6:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/networks/networks.selectors.ts:5:31
>  ❯ async /Users/petew/src/leather/extension/src/app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks.ts:7:31
>  ❯ async /Users/petew/src/leather/extension/src/app/query/bitcoin/ordinals/inscriptions.query.ts:8:31
>  ❯ async /Users/petew/src/leather/extension/src/app/query/bitcoin/ordinals/inscriptions.hooks.ts:3:31
>  ❯ async /Users/petew/src/leather/extension/src/app/query/bitcoin/ordinals/inscriptions.spec.ts:2:31
> 
> ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
> Serialized Error: { code: 'ERR_REQUIRE_ESM' }
> ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/4]⎯
> 
>  Test Files  4 failed | 32 passed (36)
> ```
> 
> I can see that got updated when re-adding `prism` [here](https://github.com/leather-wallet/extension/pull/4490/commits/c865dd40f5ebb37b30079c7dfd16b8da89d24b6c). I'm trying to think how we can solve it. We don't use `nanoid`
> 
> Here's a [SO](https://stackoverflow.com/questions/72597602/nanoid4-in-codecept-error-err-require-esm-require-of-es-module)

I implemented a fix for this which gets the tests passing https://github.com/leather-wallet/extension/pull/4522/commits/d8d03a0cf63dc05712be327b261d40dc682afef9#r1394400880

It's disabling `{ devToolsEnhancer } from '@redux-devtools/remote'` though

fbwoolf commented:
@pete-watters I got a lot more fixed today and all the tests passing atm. Once we are ready to merge, I can squash the commits down and rebase on dev. Check my commit msgs for what I fixed.

pete-watters commented on package.json:
```diff
@@ -325,6 +325,7 @@
     "webpack-shell-plugin": "0.5.0"
   },
   "resolutions": {
+    "nanoid": "3.3.4",
```
Nice, I didn't know I could just do that 😄 good to learn!

pete-watters commented on src/app/pages/onboarding/set-password/components/password-bars.tsx:
```diff
@@ -1,24 +1,26 @@
-            borderRadius="2px"
-            flexGrow={1}
+            marginTop="0"
+            style={{
```
Great 👍

pete-watters commented on src/app/features/collectibles/components/collectible-item.layout.tsx:
```diff
@@ -60,9 +61,14 @@ export function CollectibleItemLayout({
             left="0px"
             overflow="hidden"
             position="absolute"
+            style={{
```
Maybe we can write a short function for this?

Double ternary can be hard to read and could even be banned by eslint

It works though!

pete-watters reviewed (approved):
Great work on the last pieces and in general 👍

Code LGTM. I'll keep testing but would be happy to squash, rebase and merge at this point. 

We could fix any bugs in a follow up then. I'm pretty confident we tested most areas

pete-watters commented:
@kyranjamie any chance you could give this a re-review if you have time? We are code complete now but it would be good to make any final improvements now. Thanks!

mica000 commented:
Design QA [here](https://www.notion.so/trustmachines/QA-Removal-of-Stacks-UI-2381e57816714513a0ecc7702f54fb84)

fbwoolf commented:
@pete-watters I got all the commits squashed here and rebased on dev. I ran into some tests failing when the tests try to switch to testnet, but the failing has come and gone so I'm not sure if api related. You might try rerunning them in your AM. If you can get them passing, I would merge this! 🚀

kyranjamie commented on src/app/common/hooks/use-modifier-key.ts:
```diff
@@ -1,5 +1,13 @@
 import { useCallback, useEffect, useState } from 'react';
 
+export function buildEnterKeyEvent(onClick: () => void) {
+  return (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter' && onClick) {
+      onClick();
+    }
+  };
+}
```
No change needed here but this is a perfect case for a factory fn

kyranjamie commented on src/app/common/transactions/bitcoin/utils.ts:
```diff
@@ -1,14 +1,35 @@
 import { BtcSizeFeeEstimator } from './fees/btc-size-fee-estimator';
 
+type BtcTxStatus = 'pending' | 'success';
+type BtcStatusColorMap = Record<BtcTxStatus, string>;
```
We have accces to token types, so we can type more strongly than just `string`

pete-watters replied:
We took this in from `stacks-ui`, we can improve it

kyranjamie commented on src/app/components/crypto-assets/bitcoin/brc20-token-asset-list/components/brc20-token-asset-item.layout.tsx:
```diff
@@ -1,58 +1,59 @@
-              {title}
+  return (
+    // #4476 TODO not sure why this isn't erroring
+    <Flex as={isPressable ? 'button' : 'div'} outline={0} {...(bind as any)}>
```
because of the any?

pete-watters replied:
I think `<Flex as=` should have an error now as it shouldn't be OK with `leather-styles/jsx`

<img width="194" alt="Screenshot 2023-11-17 at 11 02 43" src="https://github.com/leather-wallet/extension/assets/2938440/a6c0068b-8f82-4345-bd91-5ec3946eb426">

kyranjamie commented on src/app/ui/components/icons/arrow-down-icon.tsx:
```diff
@@ -0,0 +1,25 @@
+
+import { Svg } from '@app/ui/components/svg';
+
+export function ArrowDownIcon({ size = token('icons.icon.sm'), ...props }: SquareProps) {
+  return (
+    <Square size={size} {...props}>
+      <Svg
+        xmlns="http://www.w3.org/2000/svg"
+        width="16"
+        height="16"
+        viewBox="0 0 16 16"
+        fill="none"
+      >
+        <path
+          d="M7.99992 3.33325V11.9999M3.33325 7.99992L7.99992 12.6666L12.6666 7.99992"
+          stroke="currentColor"
+          strokeWidth="1.8"
+          strokeLinecap="square"
+        />
+      </Svg>
+    </Square>
+  );
+}
```
Would make a lot more sense to have a single Arrow component accepting a direction prop and just rotating it

kyranjamie reviewed (approved):
Great work both, a bit too much to check it all thoroughly but LGTM.

Let's upgrade panda to 0.18.3 in this PR too

pete-watters commented:
> Great work both, a bit too much to check it all thoroughly but LGTM.
> 
> Let's upgrade panda to 0.18.3 in this PR too

Thanks for checking @kyranjamie 👍 

I'll upgrade to panda in a smaller PR, I didn't want to do it here and delay merge. Also @fbwoolf fixed a lot of the design issues so I don't want to risk doing that again updating panda. 

I've made note of all your feedback above and we will make improvements soon.

pete-watters commented:
> @pete-watters I got all the commits squashed here and rebased on dev. I ran into some tests failing when the tests try to switch to testnet, but the failing has come and gone so I'm not sure if api related. You might try rerunning them in your AM. If you can get them passing, I would merge this! 🚀

Great work here @fbwoolf 👏 I just re-ran that pipeline 3 more times until it passed. It's less than ideal to have it so unreliable but we can fix that in future.

fbwoolf commented:
@pete-watters thanks for getting this merged!

markmhendrickson commented:
@314159265359879 mind helping me do a general QA of this build with an eye on any UI-related inconsistencies it may have incurred as part of a larger refactor? 🙏

314159265359879 commented:
@markmhendrickson certainly, will give it a good look tomorrow.
